File: contractstudy/src/test/java/contractstudy/collectContracts/annotation/java/testData/AnnotationsArtefact.java
package contractstudy.collectContracts.annotation.java.testData;

import android.annotation.SuppressLint;

import javax.validation.constraints.Max;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.ArrayList;
import java.util.List;

public class AnnotationsArtefact {

  @NotNull
  private final String manufacturer;

  public AnnotationsArtefact(String manufacturer) {
    this.manufacturer = manufacturer;
  }

  @SuppressLint({"SetJavaScriptEnabled", "NewApi", "ShowToast"})
  public String test123(String manufacturer) {
    return manufacturer;
  }

  public String get(@Max(5) int index) { // Method Parameter Artefact
    return "string";
  }

  @Size(min = 1)
  public List<String> addProductToList() {
    return new ArrayList<>();
  }
}

File: contractstudy/src/test/java/contractstudy/collectContracts/annotation/kotlin/testData/AnnotationsArtefact.kt
package contractstudy.collectContracts.annotation.kotlin.testData

import android.annotation.SuppressLint
import javax.validation.constraints.*

class AnnotationsArtefact(
    @NotNull private val manufacturer: String
) {
    @Min(1)
    private val manufacturer123: String = "test"

    operator fun get(@Max(5) index: Int): String {
        return "string"
    }

    @SuppressLint("SetJavaScriptEnabled", "NewApi", "ShowToast")
    fun test123(manufacturer: String?): String? {
        return manufacturer
    }

    @Size(min = 1)
    fun addProductToList(): List<String> {
        return ArrayList()
    }

    fun testThis(hello: String, @Null test: Int): List<String> {
        return ArrayList()
    }
}